File: android/src/main/kotlin/com/dns_technologies/mlkit_scanner/PluginConstants.kt
package com.dns_technologies.mlkit_scanner

/** Contains method channel names used by the Dart and Android plugin sides. */
internal object PluginConstants {
    /** Shared tag used by Android scanner components. */
    const val LOG_TAG = "MLKIT_SCANNER_PLUGIN"

    /** Argument key identifying the logical widget that owns a command or event. */
    const val viewIdArgument = "viewId"

    /** Event key containing the recognized barcode payload. */
    const val barcodeArgument = "barcode"

    /** Argument key containing a camera-control value. */
    const val valueArgument = "value"

    /** Argument key containing a scan cooldown. */
    const val delayArgument = "delay"

    /** Argument key containing a recognition area. */
    const val cropRectArgument = "cropRect"

    /** Method channel name used for scanner commands and events. */
    const val channelName = "mlkit_channel"

    /** Command applying an absolute camera zoom ratio. */
    const val setZoomRatioMethod = "setZoomRatio"

    /** Command applying the desired torch state. */
    const val toggleFlashMethod = "toggleFlash"

    /** Command enabling recognition for the prepared result endpoint. */
    const val startScanMethod = "startScan"

    /** Command stopping recognition and closing its result endpoint. */
    const val cancelScanMethod = "cancelScan"

    /** Command updating the cooldown after successful recognition. */
    const val setScanDelayMethod = "setScanDelay"

    /** Command updating recognition geometry within the preview. */
    const val setCropAreaMethod = "setCropAreaMethod"

    /** Stops native work; Dart decides whether to retain or release controller ownership. */
    const val pauseCameraMethod = "pauseCameraMethod"

    /** Selects a preview and applies Dart settings on startup, resume or ownership transfer. */
    const val resumeCameraMethod = "resumeCameraMethod"

    /** Method name used to deliver recognized barcode values to Dart. */
    const val scanResultMethod = "onScanResult"
}

File: ios/Classes/PluginConstants.swift
//
//  PluginConstants.swift
//  mlkit_scanner
//
//  Created by ООО "ДНС Технологии" on 02.03.2021.
//

import Foundation

/// Constants of the Plugin.
class PluginConstants {
    /// Shared method channel name.
    static let channelName = "mlkit_channel"
    /// Argument key identifying a logical Flutter widget.
    static let viewIdArgument = "viewId"
    /// Event key containing a recognized barcode.
    static let barcodeArgument = "barcode"
    /// Event key containing a scalar event value.
    static let valueArgument = "value"
    /// Argument key containing the scan cooldown.
    static let delayArgument = "delay"
    /// Argument key containing a recognition area.
    static let cropRectArgument = "cropRect"
    /// Argument key containing a recognition type.
    static let typeArgument = "type"
    /// Method name for absolute zoom changes.
    static let setZoomRatioMethod = "setZoomRatio"
    /// Method name for desired torch state changes.
    static let toggleFlashMethod = "toggleFlash"
    /// Method name for starting barcode recognition.
    static let startScanMethod = "startScan"
    /// Method name for cancelling barcode recognition.
    static let cancelScanMethod = "cancelScan"
    /// Method name for changing the recognition cooldown.
    static let setScanDelayMethod = "setScanDelay"
    /// Method name for updating the normalized recognition area.
    static let setCropAreaMethod = "setCropAreaMethod"
    /// Stops native work; Dart decides whether to retain or release controller ownership.
    static let pauseCameraMethod = "pauseCameraMethod"
    /// Selects a preview and applies Dart settings on startup, resume or ownership transfer.
    static let resumeCameraMethod = "resumeCameraMethod"
    /// Method name for invoking Flutter Side method with success recognitions.
    static let scanResultMethod = "onScanResult"
    /// Method name for inform flutter side when torch change state
    static let changeTorchStateMethod = "changeTorchStateMethod"
    /// Method name for getting available iOS cameras.
    static let getIosAvailableCamerasMethod = "getIosAvailableCameras"
}
